File: benchmarks/fib.lua
-- Plain naive fib recursion (fair cross-lang with examples/perf/fib.hy).
local function fib(n)
    if n <= 2 then
        return 1
    end
    return fib(n - 1) + fib(n - 2)
end

print(fib(32))

File: benchmarks/fib.js
// Plain naive fib recursion (fair cross-lang with examples/perf/fib.hy).
function fib(n) {
    if (n <= 2) {
        return 1;
    }
    return fib(n - 1) + fib(n - 2);
}

console.log(fib(32));
